### PR TITLE
fix golang-cilint run

### PR DESCRIPTION
**Pull request by QvodSoldier**
#### What type of PR is this?
/kind bug

#### What this PR does / why we need it:
fix cilint run error:  context loading failed: no go files to analyze

https://github.com/golangci/golangci-lint-action/issues/23

https://github.com/argoproj/argo-rollouts/pull/579/commits/1acf99b59678840be1e72b7151e4c8fc97407cad#diff-678682767f2477de3e3c546746f8568b9a1942b2c647d32331d7e774b8ff8d9f

#### Specified Reivewers:
@Effet @sfwn

### CODE REVIEW

File: .github/workflows/ci-it.yml [hunk 1]
```diff
@@ -9,16 +9,12 @@ on:
       - release/*
   pull_request:
 jobs:
-  CI:
+  lint-go:
+    name: Lint Go code
```
Comment (liuhaoyang) [inline .github/workflows/ci-it.yml:13]: Please rename `Lint Go code` to `CI-Lint`

**Review by liuhaoyang — CHANGES_REQUESTED** [review]

**Review by liuhaoyang — COMMENTED** [review]
I think we only need to set skip-pkg-cache and skip-build-cache to true in the golangci-lint step, there is no need to split lint and build into 2 jobs.

**Comment by QvodSoldier** [comment]
> I think we only need to set skip-pkg-cache and skip-build-cache to true in the golangci-lint step, there is no need to split lint and build into 2 jobs.

**Comment by QvodSoldier** [comment]
> I think we only need to set skip-pkg-cache and skip-build-cache to true in the golangci-lint step, there is no need to split lint and build into 2 jobs.

yes, but still execute 'go mod tidy' before 'golang-cilint'

**Review by liuhaoyang — APPROVED** [review]

**Pull request merged**